lib/Controller/FoldersController.php

```diff
@@ -234,4 +234,21 @@ public function create(int $accountId, string $name) {
 
 		return new JSONResponse($this->mailManager->createFolder($account, $name));
 	}
+
+	/**
+	 * @NoAdminRequired
+	 * @TrapError
+	 *
+	 * @throws ClientException
+	 * @throws ServiceException
+	 */
+	public function subscribeFolder(int $accountId, string $folderId, bool $subscribed) {
+		$account = $this->accountService->find($this->currentUserId, $accountId);
+
+		if (empty($accountId) || empty($folderId)) {
+			return new JSONResponse(null, Http::STATUS_BAD_REQUEST);
+		}
+
+		return new JSONResponse($this->mailManager->subscribeFolder($account, base64_decode($folderId), $subscribed));
```

Comment:
See https://github.com/nextcloud/mail/pull/3144/checks?check_run_id=697235572. This returns `void`, so it can't be used for the response data. You could just call the method and afterwards return a response with `null` as data.

Alternatively, and more tolerant to faults is that this `subscribeFolder` returns the new subscription state. Like we could transport back if the change was not performed by the IMAP server for whatever reason.

Reply:
Now the subscription status is returned from backend subscribeFolder method and the frontend is updated accordingly.

Reply:
The constructor of `JSONResponse` expects an array. can we change the boolean to something like

```
[
    'subscribed' => true,
]
```
?